optimizations.qmd

```diff
@@ -1,90 +1,860 @@
+#### History
 
-- Overview of model optimization techniques for efficient AI
-- Motivations: reduce model size, latency, power consumption, etc.
-- Optimization approaches: pruning, quantization, efficient architectures, etc.
+Historically, the idea of quantization is not novel and can be traced back to ancient times, particularly in the realm of music and astronomy. In music, the Greeks utilized a system of tetrachords, segmenting the continuous range of pitches into discrete notes, thereby quantizing musical sounds. In astronomy and physics, the concept of quantization was present in the discretized models of planetary orbits, as seen in the Ptolemaic and Copernican systems.
 
-## Quantization {#sec-quant}
+During the 1800s, quantization-based discretization was used to approximate the calculation of integrals, and further used to investigate the impact of rounding errors on the integration result. However, the term "quantization" was firmly embedded in scientific literature with the advent of quantum mechanics in the early 20th century, where it was used to describe the phenomenon that certain physical properties, such as energy, exist only in discrete, quantized states. This principle was pivotal in explaining phenomena at the atomic and subatomic levels. In the digital age, quantization found its application in signal processing, where continuous signals are converted into a discrete digital form, and in numerical algorithms, where computations on real-valued numbers are performed with finite-precision arithmetic.
 
-Explanation: Quantization is a critical technique in model optimization, helping to reduce the computational and memory demands of AI models without substantially sacrificing their performance. Through various methods and schemas, it facilitates the deployment of deep learning models on embedded devices with limited resources.
+Extending upon this second application and relevant to this section, it is used in computer science to optimize neural networks by reducing the precision of the network weights. Thus, quantization, as a concept, has been subtly woven into the tapestry of scientific and technological development, evolving and adapting to the needs and discoveries of various epochs.
 
-- Motivation for model quantization
-- Post-training quantization
-- Quantization-aware training
-- Handling activations vs weights
-- Quantization schemas: uniform, mixed, per-channel
-- Quantization in practice: deployment frameworks 
+#### Initial Breakdown
 
-## Pruning {#sec-pruning}
+We begin our foray into quantization with a brief analysis of one important use for quantization.
```

Comment:
I know that this section mainly focuses on the memory/efficiency benefits of quantization, but I think one important note is that recent work has found that quantization can also help with interpretability. An example of this is [Disentanglement via Latent Quantization](https://arxiv.org/pdf/2305.18378.pdf), which builds off of [Neural Discrete Representation Learning](https://arxiv.org/pdf/1711.00937.pdf). This paper claims that "latent quantization improves the modularity and explicitness of learned representations."

Reply:
@ushabhalla thanks for the good suggestion. Do you think we should put that here or should we put that in some place where we have Explainable AI - we don't have a chapter on that because we are mostly focused on MLsys. Thanks for pointing these papers out, I learned something new!

Reply:
Hi Professor! Personally, I think that either/both makes sense! If you weren't planning to add a dedicated "Explainable AI" chapter, you could definitely incorporate a lot of those elements into the "Responsible AI" chapter. Particularly because a large use-case of TinyML is embedded systems for healthcare applications, explainability seems highly relevant.